File: test_regress/t/t_sc_names.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2020 by Edgar E. Iglesias. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(simulator => 1);

if (!$Self->have_sc) {
    skip("No SystemC installed");
}
else {
    top_filename("t/t_sc_names.v");

    compile(
        make_main => 0,
        verilator_flags2 => ["-sc --exe $Self->{t_dir}/t_sc_names.cpp"],
        );

    execute(
        check_finished => 1,
        );
}

ok(1);
1;

File: test_regress/t/t_sc_names.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('simulator')

if not test.have_sc:
    test.skip("No SystemC installed")

test.compile(make_main=False, verilator_flags2=["-sc --exe", test.pli_filename])

test.execute()

test.passes()
